exporter/collector/googlemanagedprometheus/extra_metrics.go

```diff
@@ -51,6 +52,36 @@ func AddTargetInfo(m pmetric.Metrics) pmetric.ResourceMetricsSlice {
 	return resourceMetricSlice
 }
 
+// AddScopeInfoMetric adds the otel_scope_info metric to a Metrics slice as specified in
+// https://github.com/open-telemetry/opentelemetry-specification/blob/v1.16.0/specification/compatibility/prometheus_and_openmetrics.md#instrumentation-scope-1
+func AddScopeInfoMetric(m pmetric.Metrics) pmetric.ResourceMetricsSlice {
+	resourceMetricSlice := pmetric.NewResourceMetricsSlice()
+	rms := m.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		rm.Resource().CopyTo(resourceMetricSlice.AppendEmpty().Resource())
+
+		sms := rms.At(i).ScopeMetrics()
+		for j := 0; j < sms.Len(); j++ {
+			sm := sms.At(j)
+
+			scopeInfoMetric := resourceMetricSlice.At(i).ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+			scopeInfoMetric.SetName("otel_scope_info")
+
+			dataPoint := scopeInfoMetric.SetEmptyGauge().DataPoints().AppendEmpty()
+			dataPoint.SetIntValue(1)
+
+			dataPoint.Attributes().PutStr("otel_scope_name", sm.Scope().Name())
+			dataPoint.Attributes().PutStr("otel_scope_version", sm.Scope().Version())
```

Comment:
if name/version aren't present, should we just not add these labels? wasn't sure if that would be compatible with other metrics that do have them

Reply:
We should still add them.  When doing a join between otel_scope_info and other metrics, the empty scope should still end up matching the metrics contained within the empty scope.

Reply:
that's what I thought, but wanted to check 👍